engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -87,22 +88,23 @@ public static Type create(
       boolean isProjectPrivate) {
     var eigentype = new Type(name + ".type", definitionScope, any, null, builtin, isProjectPrivate);
     var result = new Type(name, definitionScope, supertype, eigentype, builtin, isProjectPrivate);
-    result.generateQualifiedAccessor();
+    result.generateQualifiedAccessor(lang);
     return result;
   }
 
   public static Type noType() {
     return new Type("null", null, null, null, false, false);
   }
 
-  private void generateQualifiedAccessor() {
-    var node = new ConstantNode(null, this);
+  private void generateQualifiedAccessor(EnsoLanguage lang) {
+    assert lang != null;
+    var node = new ConstantNode(lang, getDefinitionScope(), this);
     var schemaBldr =
         FunctionSchema.newBuilder()
             .argumentDefinitions(
                 new ArgumentDefinition(
                     0, "this", null, null, ArgumentDefinition.ExecutionMode.EXECUTE));
-    if (isProjectPrivate) {
+    if (isProjectPrivate()) {
```

Comment:
`isProjectPrivate()` is always false. Why not just remove these three lines? An remove the whole `Type.isProjectPrivate` method. I thought that we agreed that _project private_ type does not make sense, no?

Reply:
I kept the `isProjectPrivate()` call there, so in the future it is enough to just change the implementation of the method and we'd get _private types_ with all necessary checks:

- when I created the PR there still was
- #8835 
- and it was open

>  we agreed that project _private type_ does not make sense
 
- I don't think I ever agreed to that
- having `private type InternalData` still seems OK to me
- we just don't have it yet